debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSchema.java

```diff
@@ -89,13 +92,22 @@
      */
     public MySqlSchema(Configuration config, String serverName, Predicate<String> gtidFilter, boolean tableIdCaseInsensitive, TopicSelector topicSelector) {
         this.filters = new Filters(config);
-        this.ddlParser = new MySqlDdlParser(false);
         this.tables = new Tables(tableIdCaseInsensitive);
-        this.ddlChanges = new DdlChanges(this.ddlParser.terminator());
-        this.ddlParser.addListener(ddlChanges);
         this.topicSelector = topicSelector;
         this.tableIdCaseInsensitive = tableIdCaseInsensitive;
 
+        String ddlParsingModeStr = config.getString(MySqlConnectorConfig.DDL_PARSER_MODE);
+        DdlParsingMode parsingMode = DdlParsingMode.parse(ddlParsingModeStr, MySqlConnectorConfig.DDL_PARSER_MODE.defaultValueAsString());
+
+        try {
+            this.ddlParser = parsingMode.getParserClass().newInstance();
```

Comment:
How about simply having `parsingMode.getNewParserInstance()` instead of using reflection?

Reply:
What do you mean by that? Like having a method on enum for creating a new instance by enum value?

Reply:
Nope, you can have instance methods for the specific enum values. I'll do that and push.